src/Core/Math/LinearAlgebra.hpp

```diff
@@ -46,20 +46,32 @@ template <typename Vector>
 inline Vector trunc( const Vector& v );
 
 /// Component-wise clamp() function on a floating-point vector.
-template <typename Vector>
-inline Vector clamp( const Vector& v, const Vector& min, const Vector& max );
 
+template <typename Derived>
+inline Eigen::MatrixBase<Derived> clamp( const Eigen::MatrixBase<Derived>& v,
+                                         const Eigen::MatrixBase<Derived>& min,
+                                         const Eigen::MatrixBase<Derived>& max );
```

Comment:
why changing from `Vector` to `Eigen::MatrixBase<Derived` ?

Reply:
Here again, it's a mater of template parameter, when using Ra::Core::Vector, you can't differentiate `clamp(float, float, float)`, `clamp(vector, float, float)` and clamp `(vector, vector, vector)`
(the change comes from the fact that this clamp function have been move to the same namespace as `clamp(T,T,T)` ...)